switchtec.c

```diff
@@ -197,6 +205,9 @@ static void mrpc_complete_cmd(struct switchtec_dev *stdev)
 	stuser = list_entry(stdev->mrpc_queue.next, struct switchtec_user,
 			    list);
 
+	if (stuser->state == MRPC_ERROR)
+		goto out;
```

Comment:
Instead of this run-around, why not put the complete stuff below in a static function and call it directly in the error path.

Reply:
Updated. Please comment.